src/ray/raylet/agent_manager.cc

```diff
@@ -65,14 +67,16 @@ void AgentManager::StartAgent() {
   ProcessEnvironment env;
   env.insert({"RAY_NODE_ID", options_.node_id.Hex()});
   env.insert({"RAY_RAYLET_PID", std::to_string(getpid())});
+  // Report the restart count to the agent so that we can decide whether or not
+  // report the error message to drivers.
+  env.insert({"RESTART_COUNT", std::to_string(agent_restart_count_)});
+  env.insert({"MAX_RESTART_COUNT",
+              std::to_string(RayConfig::instance().agent_max_restart_count())});
   Process child(argv.data(), nullptr, ec, false, env);
   if (!child.IsValid() || ec) {
     // The worker failed to start. This is a fatal error.
     RAY_LOG(FATAL) << "Failed to start agent with return value " << ec << ": "
                    << ec.message();
-    RAY_UNUSED(delay_executor_([this] { StartAgent(); },
-                               RayConfig::instance().agent_restart_interval_ms()));
-    return;
```

Comment:
Is it safe to access the invalid child?

Reply:
oh we don't need this part because when the FATAL logs are printed raylet will crash anyway

Reply:
Sorry, I didn't notice the log level.